circuits/cpp/src/aztec3/circuits/rollup/test_utils/utils.cpp

```diff
@@ -130,6 +138,62 @@ BaseRollupInputs base_rollup_inputs_from_kernels(std::array<KernelData, 2> kerne
     baseRollupInputs.new_commitments_subtree_sibling_path =
         get_sibling_path<PRIVATE_DATA_SUBTREE_INCLUSION_CHECK_DEPTH>(private_data_tree, 0, PRIVATE_DATA_SUBTREE_DEPTH);
 
+    // Update public data tree to generate sibling paths: we first set the initial public data tree to the result of all
+    // state reads and old_values from state transitions. Note that, if the right tx reads or writes an index that was
+    // already processed by the left one, we don't want to reflect that as part of the initial state, so we skip those.
+    std::set<size_t> visited_indices;
```

Comment:
It doesn't look like you need this for anything other than remembering visited indices so I would consider using an unordered_set instead. Insertions and lookups will be constant time instead of logarithmic.

Reply:
I'd argue that being a test util, and with the total number of indices being at most 16, there's no really much of a need to optimize. But it's an easy change, and I appreciate learning more about the std!